google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITBackupTest.java

```diff
@@ -349,14 +352,79 @@ public void testBackups() throws InterruptedException, ExecutionException {
     testGetBackup(db2, backupId2, expireTime);
     testUpdateBackup(backup1);
     testCreateInvalidExpirationDate(db1);
-    testRestore(backup1, op1);
+    testRestore(backup1, op1, versionTime);
 
     testDelete(backupId2);
     testCancelBackupOperation(db1);
     // Finished all tests.
     logger.info("Finished all backup tests");
   }
 
+  @Test(expected = SpannerException.class)
+  public void backupCreationWithVersionTimeTooFarInThePastFails() throws Exception {
+    final Database testDatabase = testHelper.createTestDatabase();
+    final DatabaseId databaseId = testDatabase.getId();
+    final InstanceId instanceId = databaseId.getInstanceId();
+    final String backupId = testHelper.getUniqueBackupId();
+    final Timestamp expireTime = afterDays(7);
+    final Timestamp versionTime = daysAgo(30);
+    final Backup backupToCreate =
+        dbAdminClient
+            .newBackupBuilder(BackupId.of(instanceId, backupId))
+            .setDatabase(databaseId)
+            .setExpireTime(expireTime)
+            .setVersionTime(versionTime)
+            .build();
+
+    getOrThrow(dbAdminClient.createBackup(backupToCreate));
+  }
+
+  @Test(expected = SpannerException.class)
+  public void backupCreationWithVersionTimeInTheFutureFails() throws Exception {
+    final Database testDatabase = testHelper.createTestDatabase();
+    final DatabaseId databaseId = testDatabase.getId();
+    final InstanceId instanceId = databaseId.getInstanceId();
+    final String backupId = testHelper.getUniqueBackupId();
+    final Timestamp expireTime = afterDays(7);
+    final Timestamp versionTime = afterDays(1);
+    final Backup backupToCreate =
+        dbAdminClient
+            .newBackupBuilder(BackupId.of(instanceId, backupId))
+            .setDatabase(databaseId)
+            .setExpireTime(expireTime)
+            .setVersionTime(versionTime)
+            .build();
+
+    getOrThrow(dbAdminClient.createBackup(backupToCreate));
+  }
+
+  private <T> T getOrThrow(OperationFuture<T, ?> operation)
+      throws InterruptedException, ExecutionException {
+    try {
+      return operation.get();
+    } catch (ExecutionException e) {
+      if (e.getCause() != null && e.getCause() instanceof SpannerException) {
```

Comment:
nit: null-check is redundant, as `null instanceof whatever` always returns false

Reply:
Thanks, fixed it.